core/src/main/java/org/mule/runtime/core/internal/exception/GlobalErrorHandler.java

```diff
@@ -41,56 +32,61 @@ public Publisher<CoreEvent> apply(Exception exception) {
 
   @Override
   public void initialise() throws InitialisationException {
-    setFromGlobalErrorHandler();
-    super.initialise();
-  }
-
-  private void setFromGlobalErrorHandler() {
-    this.getExceptionListeners().stream()
-        .filter(exceptionListener -> exceptionListener instanceof TemplateOnErrorHandler)
-        .forEach(exceptionListener -> ((TemplateOnErrorHandler) exceptionListener).setFromGlobalErrorHandler(true));
+    if (featureFlaggingService.isEnabled(REUSE_GLOBAL_ERROR_HANDLER)) {
```

Comment:
This can be reduced to a single and in the condition you can verify the featureflagservice, the atomic integer or return true when appropriate. In this way you avoid the nested if and it makes the code clear.